File: src/main/java/io/sch/historyscan/domain/contexts/codebase/find/CodeBaseSearch.java
package io.sch.historyscan.domain.contexts.codebase.find;

import java.util.Optional;

public record CodeBaseSearch(CodeBaseInfoInventory codeBaseInfoInventory) implements FindCodeBase {

    @Override
    public Optional<CurrentCodeBase> from(String name) {
        return codeBaseInfoInventory.findBy(name);
    }
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/codebase/find/CodeBaseSearch.kt
package io.sch.historyscan.domain.contexts.codebase.find

class CodeBaseSearch(private val codeBaseInfoInventory: CodeBaseInfoInventory) : FindCodeBase {
    override fun from(name: String): CurrentCodeBase? {
        return codeBaseInfoInventory.findBy(name)
    }
}
